src/content/en/fundamentals/building-components/best-practices.md

```diff
@@ -0,0 +1,535 @@
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+<table class="responsive ce-cl">
+  <tbody>
+    <tr>
+      <th colspan=2>
+        <h3>
+  Do not self-apply classes.
+        </h3>
+      </th>
+    </tr>
+    <tr>
+      <td><b>Why?</b></td>
+      <td>
+  Elements that need to express their state should do so using attributes. The
+  <code>class</code> attribute is generally considered to be owned by the
+  developer using your element, and writing to it yourself may inadvertantly
+  stomp on developer classes.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+### Events
+
+<table class="responsive ce-cl">
+  <tbody>
+    <tr>
+      <th colspan=2>
+        <h3>
+  Dispatch events in response to internal component activity.
+        </h3>
+      </th>
+    </tr>
+    <tr>
+      <td><b>Why?</b></td>
+      <td>
+  Your component may have properties that change in response to activity that
+  only your component knows about. For example, if a timer or animation
+  finishes, or a resource has finished loading. It's helpful to dispatch events
+  in response to these changes to notify the host that the component's state is
+  different.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+<table class="responsive ce-cl">
+  <tbody>
+    <tr>
+      <th colspan=2>
+        <h3>
+  Do not dispatch events in response to the host setting a property (downward
+  data flow).
+        </h3>
+      </th>
+    </tr>
+    <tr>
+      <td><b>Why?</b></td>
+      <td>
+  Dispatching an event in response to a host setting a property is superfluous
+  (the host knows the current state because it just set it). Dispatching events
+  in response to a host setting a property may cause infinite loops with data
+  binding systems.
+      </td>
+    </tr>
+    <tr>
+      <td><b>Example</b></td>
+      <td>
+  The <a href="/web/fundamentals/building-components/examples/howto-checkbox">
+  <code>&lt;howto-checkbox&gt;</code></a> element.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+## Explainers {: #explainers }
+
+### Don't override the page author {: #dont-override }
+
+It's possible that a developer using your element might want to override some of
+its initial state. For example, changing its ARIA `role` or focusablity with
+`tabindex`. Check to see if these and any other global attributes have been set,
+before applying your own values.
+
+```js
+connectedCallback() {
+  if (!this.hasAttribute('role'))
+    this.setAttribute('role', 'checkbox');
+  if (!this.hasAttribute('tabindex'))
+    this.setAttribute('tabindex', 0);
+```
+
+### Make properties lazy {: #lazy-properties }
+
+A developer might attempt to set a property on your element before its
+definition has been loaded. This is especially true if the developer is using a
+framework which handles loading components, stamping them to the page, and
+binding their properties to a model.
+
+```html
+<!--
+Here Angular is declaratively binding its model's isChecked property to the
```

Comment:
since this is inside of an HTML comment I'm not sure if the backticks will work but I can try